pages/api/VolunteerRoleService.ts

```
@@ -0,0 +1,187 @@
+      })
+
+      return volunteerRolesData
+    } catch (err) {
+      console.log(err)
+      throw err
+    }
+  }
+  async getRoleDetailsByName(
+    roleKey: string
+  ): Promise<DASVolunteerRole | null> {
+    try {
+      const recordData = await this.volunteerRolesTable
+        .select({
+          view: 'Grid view',
+          maxRecords: 1,
+          filterByFormula: `{Key} = '${roleKey}'`,
+        })
+        .firstPage()
+      if (recordData.length === 0) {
+        return null
+      }
+      const record = recordData[0]
+
+      if (record.fields.Status !== 'Active') {
+        return null
+      }
+      let keyTechnologies = null
+
+      if (record.fields['Key technologies'] && Array.isArray(record.fields['Key technologies'])) {
+        const technologiesRecordIds:Array<any> = record.fields['Key technologies']
+        try {
+          const technologiesData = await this.toolsWeUseTable
+            .select({
+              view: 'Grid view',
+            })
+            .all()
+          const roleTechnologies = technologiesData.filter((tech) => {
+            return technologiesRecordIds.includes(tech.id)
+          })
+          const technologiesMap = new Map()
+          roleTechnologies.forEach((tech) => {
+            const category = tech.fields?.Category || 'Other'
+            const name = tech.fields?.['Name']
+            if (!technologiesMap.has(category)) {
+              technologiesMap.set(category, [])
+            }
+            technologiesMap.get(category).push(name)
+          })
+          keyTechnologies = []
+          technologiesMap.forEach((value, key) => {
+            keyTechnologies.push(`${key}: ${value.join(', ')}`)
+          })
+          keyTechnologies = keyTechnologies.sort((a, b) => {
+            if (a.startsWith('Other')) {
+              return 1
+            } else if (b.startsWith('Other')) {
+              return -1
+            } else {
+            }
+          })
+        } catch (error) {}
```

bad...swallowing errors

---

OBTW, should we start using console.error(error) instead of console.log(error)?